framework_agreement/model/purchase.py

```diff
@@ -190,14 +134,27 @@ def _propagate_fields(self):
     @api.onchange('framework_agreement_id')
     def onchange_agreement(self):
         self._propagate_fields()
-
         if self.framework_agreement_id:
             agreement = self.framework_agreement_id
             if agreement.supplier_id != self.order_id.partner_id:
-                raise exceptions.Warning(
+                raise UserError(
                     _('Invalid agreement '
-                      'Agreement and supplier does not match')
-                )
-
+                      'Agreement and supplier does not match'))
             self.price_unit = agreement.get_price(self.product_qty,
                                                   self.order_id.currency_id)
+
+    @api.multi
+    @api.constrains('price_unit')
+    def _check_line_price_unit_framework_agreement(self):
+        for record in self:
+            if record.framework_agreement_id:
+                agreement_price = record.framework_agreement_id.get_price(
+                    record.product_qty,
+                    currency=record.currency_id)
+                if agreement_price != self.price_unit:
+                    msg = _(
+                        "In line %s You have set the price to %s \n"
+                        " but there is a running agreement"
+                        " with price %s") % (
```

Comment:
Is it possible that the PO is in USD and agreement in EUR and price been the same would pass the validation?